api/pkg/transformer/symbol/time.go

```diff
@@ -0,0 +1,240 @@
+package symbol
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/gojek/merlin/pkg/transformer/symbol/function"
+	"github.com/gojek/merlin/pkg/transformer/types/converter"
+)
+
+// Now() returns current local time
+func (sr Registry) Now() time.Time {
+	return time.Now()
+}
+
+// DayOfWeek will return number represent day in a week, given timestamp and timezone
+// SUNDAY(0), MONDAY(1), TUESDAY(2), WEDNESDAY(3), THURSDAY(4), FRIDAY(5), SATURDAY(6)
+// timestamp can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) DayOfWeek(timestamp, timezone interface{}) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return function.DayOfWeek(ts, tz)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+// IsWeekend check wheter given timestamps falls in weekend, given timestamp and timezone
+// timestamp ana timezone can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) IsWeekend(timestamp, timezone interface{}) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return function.IsWeekend(ts, tz)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+// FormatTimestamp will convert timestamp into formatted date string
+// timestamp can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) FormatTimestamp(timestamp, timezone interface{}, format string) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return function.FormatTimestamp(ts, tz, format)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+// ParseTimeStamp convert timestamp value into time
+func (sr Registry) ParseTimestamp(timestamp interface{}) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return time.Unix(ts, 0).In(tz)
+	}
+	// empty timezone mean using UTC
+	return sr.processTimestampFunction(timestamp, "", timeFn)
+}
+
+func (sr Registry) processTimestampFunction(timestamps, timezone interface{}, timeTransformerFn func(timestamp int64, tz *time.Location) interface{}) interface{} {
+	ts, err := sr.evalArg(timestamps)
+	if err != nil {
+		panic(err)
+	}
+
+	tz, err := sr.evalArg(timezone)
+	if err != nil {
+		panic(err)
+	}
+
+	timestampVals := reflect.ValueOf(ts)
+	timezoneVals := reflect.ValueOf(tz)
+
+	switch timestampVals.Kind() {
+	case reflect.Slice:
+		if timestampVals.Len() == 0 {
+			panic("empty array of timestamp provided")
+		}
+
+		location := &time.Location{}
+		if timezoneVals.Kind() == reflect.Slice {
```

Comment:
can we move out the validation of timezonevals into separate function, i notice it also be used in `ParseDateTime` function ?